snap/snapcraft.yaml

```diff
@@ -13,6 +13,16 @@ base: core22
 architectures:
   - build-on: arm64
 
+slots:
+  custom-gpio-dev:
+    interface: custom-device
+    custom-device: gpiochip
+    devices:
+      - /dev/gpiochip[0-4]
+    files:
+      read:
+        - /sys/devices/platform/soc/*.gpio/gpiochip[0-4]/dev
```

Comment:
As tested, this doesn't work on Pi5 as it needs read access to `/sys/devices/platform/axi/1000120000.pcie/1f000d0000.gpio/gpiochip4/dev` instead of `/sys/devices/platform/soc/*.gpio/gpiochip[0-4]/dev`.

Moreover, there is no need to expose `gpiochip[1-3]` for GPIO control on any existing Pi.

The slot to potentially cover both legacy Pis and Pi5 could be something like:
```yaml
  custom-gpio-dev:
    interface: custom-device
    custom-device: gpiochip
    devices:
      # Legacy Raspberry Pis
      - /dev/gpiochip0
      # Raspberry Pi 5
      - /dev/gpiochip4
    files:
      read:
        - /sys/devices/platform/soc/*.gpio/gpiochip0/dev
        - /sys/devices/platform/axi/*.pcie/*.gpio/gpiochip4/dev
```